src/main/java/net/pms/dlna/DLNAThumbnail.java

```diff
@@ -0,0 +1,263 @@
+	 *         source image could not be parsed.
+	 */
+	public static DLNAThumbnail toThumbnail(InputStream inputStream) {
+		return toThumbnail(inputStream, 0, 0, null, ImageFormat.SOURCE);
+	}
+
+	/**
+	 * Converts an image to a {@link DLNAThumbnail}. Format support is limited
+	 * to that of {@link ImageIO}.
+	 *
+	 * <p><b>
+	 * This method consumes and closes {@code inputStream}.
+	 * </b>
+	 * @param inputStream the source image in a supported format.
+	 * @param width the new width or 0 to disable scaling.
+	 * @param height the new height or 0 to disable scaling.
+	 * @param scaleType the {@link ScaleType} to use when scaling.
+	 * @param outputFormat the {@link ImageFormat} to generate or
+	 *                     {@link ImageFormat#SOURCE} to preserve source format.
+	 * @return The populated {@link DLNAThumbnail} or {@code null} if the
+	 *         source image could not be parsed.
+	 */
+	public static DLNAThumbnail toThumbnail(InputStream inputStream, int width, int height, ScaleType scaleType, ImageFormat outputFormat) {
+		if (inputStream == null) {
+			return null;
+		}
+		try {
+			// A little optimization that saves us from recreating an image that
+			// is already in the correct format/size and is a byte array.
+			byte[] bytes = null;
+			if (inputStream instanceof ByteArrayInputStream) {
+				bytes = ImagesUtil.toByteArray(inputStream);
+				inputStream = new ByteArrayInputStream(bytes);
+			}
+
+			ImageReaderResult imageResult = CustomImageReader.read(inputStream);
+			inputStream.close();
+
+			if (imageResult.bufferedImage == null || imageResult.imageFormat == null) { // ImageIO doesn't support the image format
+				LOGGER.error("Failed to create thumbnail because the source format is unknown");
+				return null;
+			}
+
+			if (outputFormat == null || outputFormat.equals(ImageFormat.SOURCE)) {
+				outputFormat = imageResult.imageFormat;
+			}
+
+			// Thumbnails aren't useful in all formats, restrict them
+			if (!ImageFormat.JPEG.equals(outputFormat) && !ImageFormat.PNG.equals(outputFormat)) {
+				outputFormat = ImageFormat.JPEG;
+			}
+
+			if (width == 0 ||
+				height == 0 ||
+				(
+					ScaleType.MAX.equals(scaleType) &&
+					imageResult.bufferedImage.getWidth() <= width &&
+					imageResult.bufferedImage.getHeight() <= height
+				)
+			) {
+				//No resize, just convert
+				if (imageResult.imageFormat.equals(outputFormat) && bytes != null) {
+					// Return source
+					return new DLNAThumbnail(bytes, imageResult.bufferedImage.getWidth(), imageResult.bufferedImage.getHeight(), imageResult.imageFormat, false);
+				}
+				ByteArrayOutputStream out = new ByteArrayOutputStream();
+				ImageIO.setUseCache(false);
+				ImagesUtil.imageIOWrite(imageResult.bufferedImage, outputFormat.toString(), out);
+				return new DLNAThumbnail(out.toByteArray(), imageResult.bufferedImage.getWidth(), imageResult.bufferedImage.getHeight(), outputFormat, false);
+			} else {
+				BufferedImage bufferedImage =
```

Comment:
@Sami32 Yes. I'm not sure what part you're not sure if is ok, so I'll explain both possibilities I can think of:
* Setting a variable to nothing: It's not, the "line" continue on the next line(s). There's an old standard (from the time people coded on terminals) that says that no line should be more than 80 characters. Personally, I think that makes the code very difficult to read, so I usually use 120 characters instead. But, UMS seems to "kind of" adhere to the 80 character limit (it's broken a lot of places though), so I try to do the same. That means that you have to break up long lines, and try to do it in some kind of logical way. This is all about human readability though, as newline doesn't mean anything to the Java compiler. As it sees it, a line ends whenever it finds a ```;```, so you have have many "Java lines" on one document line, or you can have one "Java line" on many document lines.
* Using the same name for the type and the variable: Actually, it's not the same as Java source is case sensitive. The compiler wouldn't allow for them to be identical. Naming convention dictates that objects/classes/types start with an uppercase letter while variables/fields/methods should start with a lowercase letter. Except for that, they are all CamelCase. I prefer giving the variable the same name as the type and try to do it where ever it's practical. I think it makes the code much easier to read, as you can instantly tell the type when you read the code instead of having to find the declaration to know what type it is.